spec/rex/socket_spec.rb

```diff
@@ -342,6 +342,13 @@
        end
      end
 
+    context 'with a name containing underscores' do
```

Comment:
Not a blocker; would it make sense to rename `is_name?` is `is_dns_name?` or something similar? I intentionally didn't call it on the initial PR as I thought the distinctiong wasn't super importance, but now I wish I had as hostname vs DNS name have very different rules apparently 😄